### PR TITLE
backend-tasks: Introduce human friendly scheduling

### PR DESCRIPTION
**Pull request by jhaals**
Extend the `TaskScheduleDefinition` with a human friendly durations object removing the need to depend on Luxon for timeouts and cron for scheduling.

### CODE REVIEW

File: packages/backend-tasks/src/tasks/PluginTaskSchedulerImpl.ts [hunk 1]
```diff
@@ -48,6 +49,20 @@ export class PluginTaskSchedulerImpl implements PluginTaskScheduler {
...
   }
 
+  private parseDuration(
+    frequency: TaskScheduleDefinition['frequency'],
+  ): string {
+    if ('cron' in frequency) {
+      return frequency.cron;
+    }
+
+    if (frequency instanceof Duration) {
```
Comment (Rugvip) [inline packages/backend-tasks/src/tasks/PluginTaskSchedulerImpl.ts:59]: This bit

**Review by Rugvip — APPROVED** [review]
✨

File: packages/backend-tasks/src/tasks/PluginTaskSchedulerImpl.ts [hunk 2]
```diff
@@ -48,6 +49,20 @@ export class PluginTaskSchedulerImpl implements PluginTaskScheduler {
...
+  ): string {
+    if ('cron' in frequency) {
+      return frequency.cron;
+    }
+
+    if (Duration.isDuration(frequency)) {
+      return frequency.toISO();
+    }
+
+    return Duration.fromObject(frequency).toISO();
```
Comment (freben) [inline packages/backend-tasks/src/tasks/PluginTaskSchedulerImpl.ts:63]: no tests actually exercise this new code

File: packages/backend-tasks/src/tasks/PluginTaskSchedulerImpl.ts [hunk 3]
```diff
@@ -61,12 +76,11 @@ export class PluginTaskSchedulerImpl implements PluginTaskScheduler {
...
-            'cron' in task.frequency
-              ? task.frequency.cron
-              : task.frequency.toISO(),
-          initialDelayDuration: task.initialDelay?.toISO(),
-          timeoutAfterDuration: task.timeout.toISO(),
+          cadence: this.parseDuration(task.frequency),
+          initialDelayDuration:
+            task.initialDelay && this.parseDuration(task.initialDelay),
+          timeoutAfterDuration:
+            task.timeout && this.parseDuration(task.timeout),
```
Comment (freben) [inline packages/backend-tasks/src/tasks/PluginTaskSchedulerImpl.ts:83]: the timeout can actually not be undefined

File: packages/backend-tasks/src/tasks/PluginTaskSchedulerImpl.ts [hunk 4]
```diff
@@ -78,12 +92,11 @@ export class PluginTaskSchedulerImpl implements PluginTaskScheduler {
...
```
Comment (freben) [inline packages/backend-tasks/src/tasks/PluginTaskSchedulerImpl.ts:99]: same comments as above

Comment (jhaals) [reply packages/backend-tasks/src/tasks/PluginTaskSchedulerImpl.ts]: fixed

File: packages/backend-tasks/src/tasks/PluginTaskSchedulerImpl.ts (hunk 2 above)
Comment (jhaals) [reply packages/backend-tasks/src/tasks/PluginTaskSchedulerImpl.ts]: moved this out to a separate function to test it. The tests in TaskWorker.test.ts was deeper down and used Duration directly. The existing Impl test did not exercise this in its current form.

Unrelated but should this be renamed to DefaultTaskScheduler? 🤔

Comment (freben) [reply packages/backend-tasks/src/tasks/PluginTaskSchedulerImpl.ts]: There will only ever be one implementation though, and it's internal.

**Review by freben — APPROVED** [review]

**Pull request merged**